src/couch/src/couch_att.erl

```diff
@@ -179,6 +180,24 @@ merge_stubs([], _, Merged) ->
     {ok, lists:reverse(Merged)}.
 
 
+external_size(Att) ->
+    NameSize = size(fetch(name, Att)),
+    TypeSize = case fetch(type, Att) of
+        undefined -> 0;
+        Type -> size(Type)
+    end,
+    AttSize = fetch(att_len, Att),
+    Md5Size = case fetch(md5, Att) of
+        undefined -> 0;
+        Md5 -> size(Md5)
+    end,
+    HeadersSize = case fetch(headers, Att) of
+        undefined -> 0;
+        Headers -> couch_ejson_size:encoded_size(Headers)
```

Comment:
This will blow up with a function clause as headers is a proplist.

However, when I tried to make it crash but couldn't find a way to even create an attachment with headers. I don't think it is possible to with FDB. It might have been a compatibility clause with old .couch files... 

Maybe we should remove headers from `couch_att` so we don't have to carry extra baggage forward? Or just ignore it here for now to minimize the diff between 3.x and FDB layer branch?

Reply:
So I went looking for this and managed to confuse myself quite thoroughly before figuring out what happened. The previous code used headers as a way to test the upgrade between attachment records and proplists. When I rewrote the couch_att module to handle FoundationDB things I accidentally exploded the test code into a whole headers concept that exists nowhere outside of couch_att. So the correct answer here is to remove that and all related tests.

Reply:
Fixed!